PAPER.md

```diff
@@ -0,0 +1,378 @@
+|       |                                                       |
++-+-+-+-+                                                       +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                                                               |
++                                                               +
+|                        destination(256)                       |
++       +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|       |         headers len(16)       |                       |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+                       +
+|                             headers                           |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          data len(32)                         |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|                          data                         |       |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+|    signature len(16)  |                                       |
++-+-+-+-+-+-+-+-+-+-+-+-+                                       +
+|                            signature                          |
++-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+
+```
+
+* `type` field for this message will be the unsigned integer `4`.
+
+* `uuid` field its an [UUID](https://www.rfc-editor.org/rfc/rfc4122.html) v4 that needs to be added to every message by the application, so previous messages can be easy referenced and found later.
+
+* `prev hash` field will be the SHA256 sum of the last acknowledged message in the conversation.
+
+* `timestamp` field is an unsigned four-octet number containing the number of seconds elapsed since midnight, 1 January 1970 UTC. Represents the moment of the message signing, which should be close to the send time.
+
+* `source` field is the SHA256 hash value of the sender public key. Creator of the message should add its public key fingerprint here.
+
+* `destination` field is the SHA256 hash value of the receiver public key.
+
+* `headers` is a variable length field. It can be used to add extra metadata to the message in a key value fashion. The specific format for this field could be borrowed from the [HTTP headers standard](https://www.rfc-editor.org/rfc/rfc7230#section-3.2).
```

Comment:
Which metainformation do you have in mind? Could you provde an example?

Reply:
Ok, here it is:
https://github.com/eloylp/decentra-chat/pull/1#discussion_r1010331158